coap-core/src/main/java/com/mbed/coap/observe/SimpleObservableResource.java

```diff
@@ -52,16 +53,37 @@ public void get(CoapExchange exchange) throws CoapCodeException {
      * @throws CoapException coap exception
      */
     public void setBody(String body) throws CoapException {
+        this.body = body.getBytes(DEFAULT_CHARSET);
+        notifyChange(this.body, null);
```

Comment:
minor: could this call 
```setBody(body.getBytes(DEFAULT_CHARSET))```

Reply:
Why not. I guess we should do this so that no body would change this.body?